pkg/daemon/certificate_writer.go

```diff
@@ -110,6 +115,36 @@ func (dn *Daemon) syncControllerConfigHandler(key string) error {
 		pathToData[caBundleFilePath] = kubeAPIServerServingCABytes
 		pathToData[cloudCABundleFilePath] = cloudCA
 
+		cm, err := dn.clusterCmLister.ConfigMaps("openshift-config-managed").Get("kube-apiserver-server-ca")
+		if err != nil {
+			return err
+		}
+
+		data, err := cmToData(cm, "ca-bundle.crt")
+
+		if data != nil {
+			currentKC := clientcmdv1.Config{}
+			kcBytes, err := os.ReadFile(kubeConfigPath)
+			if err != nil {
+				return err
+			}
+			if kcBytes != nil {
+				err = json.Unmarshal(kcBytes, &currentKC)
+				if err != nil {
+					return fmt.Errorf("Could not unmarshal kubeconfig into struct. Data: %s, Error: %v", string(kcBytes), err)
+				}
+				var newData []byte
+				if len(currentKC.Clusters) > 0 {
+					currentKC.Clusters[0].Cluster.CertificateAuthorityData = data
+					newData, err = json.Marshal(currentKC)
+					if err != nil {
+						return fmt.Errorf("Could not marshal kubeconfig into bytes. Error: %v", err)
+					}
+				}
+				pathToData[kubeConfigPath] = newData
```

Comment:
This does not have an opinion on file mode.  Is there logic somewhere (maybe not in this backportable pull) to stomp over anyone who tries to set an overly broad mode on this secret-containing file?  On a random 4.15.0-ec.3 cluster:

```console
$ oc --as system:admin debug node/build0-gstfj-ci-tests-worker-c-br9cw -- ls -l /host/etc/kubernetes/kubeconfig /host/var/lib/kubelet/kubeconfig
Temporary namespace openshift-debug-hmt7w is created for debugging node...
Starting pod/build0-gstfj-ci-tests-worker-c-br9cw-debug-pxzqn ...
To use host binaries, run `chroot /host`
-rw-------. 1 root root 15205 Jan 10 19:52 /host/etc/kubernetes/kubeconfig
-rw-------. 1 root root 14425 Jan 10 19:56 /host/var/lib/kubelet/kubeconfig

Removing debug pod ...
Temporary namespace openshift-debug-hmt7w was removed.
```

I know #3808 added mode management to `/etc/kubernetes/kubeconfig`, but if you [pivot this code to write `/var/lib/kubelet/kubeconfig`][1], we might want to expand that mode management and don't want this code fighting with the #3808 code.

[1]: https://github.com/openshift/machine-config-operator/pull/4106#discussion_r1448040803

Reply:
good catch, I'll make sure file mode is honored. We have code in the cert writer that checks for file mode on existing files I believe.